software/bsg_manycore_lib/bsg_set_tile_x_y.h

```diff
@@ -1,3 +1,21 @@
+/* Each application launched onto the manycore is represented by a grid of tile groups.        */
+/* A grid is a two diemsnional structure represeting the entire application, with each         */
+/* element of grid being a tile group with X and Y coordinates bsg_tile_group_id_x/y,          */
+/* and the flat coordiante of bsg_tile_group_id, which is calculate using grid dimensions      */
+/* __bsg_x               --> The X coordinate of tile inside its tile group                    */
+/* __bsg_y               --> The Y coordiante of tile inside its tile group                    */
+/* __bsg_id              --> The flat index of tile inside tile group                          */
+/*                            __bsg_id = __bsg_y * __bsg_tiles_X + __bsg_x                     */
+/* __bsg_grp_org_x       --> The X coordinate of the tile group origin tile                    */
+/* __bsg_grp_org_y       --> The Y coordinate of the tile group origin tile                    */
+/* __bsg_grid_dim_x      --> X dimension of grid, or number of tile groups in the X dimensions */
```

Comment:
are you saying that "grid" is 2D array of "tile groups"?
and a tile group is a 2D array of tiles?